safe_transaction_service/contracts/tx_decoder.py

```diff
@@ -552,6 +552,34 @@ class DbTxDecoder(TxDecoder):
 
     cache_abis_by_address = TTLCache(maxsize=2048, ttl=60 * 5)  # 5 minutes of caching
 
+    @classmethod
+    def get_most_relevant_contract_abi(cls, address: ChecksumAddress):
+        """
+        Retrieves the most relevant ABI for the contract at the given address.
```

Comment:
This is working, but it's not accurate. `Contract address` is a `primary key`, so for one address there can be either `1` or `0` matching ABIs, so you don't need the sorting.

I don't know why we were doing it that way

Reply:
Updated [here](https://github.com/safe-global/safe-transaction-service/pull/2262/commits/4caf91e5a242ebb07a6d3ff2cf11d225dbefc8da)